packages/dataviews/src/dataviews-layouts/table/style.scss

```diff
@@ -297,3 +297,14 @@
 		max-width: 80ch;
 	}
 }
+
+// Group header styling
+.dataviews-view-table__group-header-row {
+	background-color: $gray-100;
+
+	.dataviews-view-table__group-header-cell {
+		font-weight: 600;
```

Comment:
We can use a [token](https://wordpress.github.io/gutenberg/?path=/docs/tokens-typography--page#weight) here :)